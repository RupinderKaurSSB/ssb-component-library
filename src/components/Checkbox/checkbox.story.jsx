import React from 'react';
import { storiesOf } from '@storybook/react';
import centered from '@storybook/addon-centered/react';
import Checkbox from './index';

storiesOf('Checkbox', module).addDecorator(centered)
	.add('Default', () => (
		<div style={{ display: 'flex' }}>
			<Checkbox value="item">Item</Checkbox>
		</div>
	));
