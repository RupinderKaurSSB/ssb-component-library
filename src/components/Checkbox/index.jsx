import React from 'react';
import PropTypes from 'prop-types';

const Checkbox = ({
	index, children, callback, selected, value, disabled,
}) => (
	<div tabIndex={index + 1} className={`checkbox-button-wrapper${disabled ? ' disabled' : ''}`}>
		<label htmlFor={value}>
			{children}
			<input
				id={value}
				value={value}
				type="checkbox"
				checked={selected}
				onChange={() => callback(value)}
			/>
			<span className="check-mark" />
		</label>
	</div>
);

Checkbox.defaultProps = {
	callback: () => {},
	index: 1,
};

Checkbox.propTypes = {
	callback: PropTypes.func,
	children: PropTypes.node,
	index: PropTypes.number,
	selected: PropTypes.bool,
	value: PropTypes.oneOfType([
		PropTypes.string,
		PropTypes.number,
	]).isRequired,
	disabled: PropTypes.bool,
};

export default Checkbox;
