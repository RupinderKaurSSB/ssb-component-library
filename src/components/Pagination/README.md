Pagination
========

> A component for navigating through pages split up in numbers

### Usage

#### HTML

```html
<div class="ssb-pagination">
  <div class="direction-button">
    <i class="chevron-icon feather-chevron-left" />Forrige
  </div>
  <div class="nav-button-square selected" onclick="{select this}">1</div>
  <div class="nav-button-square" onclick="{select this}">2</div>
  <div class="nav-button-square" onclick="{select this}">3</div>
  <div class="nav-button-square" onclick="{select this}">4</div>
  <div class="nav-button-square" onclick="{select this}">5</div>
  <div class="nav-button-square" onclick="{select this}">6</div>
  <div class="nav-button-square" onclick="{select this}">7</div>
  <div class="nav-button-square" onclick="{select this}">8</div>
  <div class="dotted-indicator">...</div>
  <div class="nav-button-square" onclick="{select this}">20</div>
  <div class="direction-button">
    Neste<i class="chevron-icon feather-chevron-left" />
  </div>
</div>
```

#### React

```jsx harmony
const items = [
  { text: '1', path: '/1' },
  { text: '2', path: '/2' },
  { text: '3', path: '/3' },
  ...
  { text: '20', path: '/20' },
];
<Pagination items={items} selectedPage={items[4]} />
```

Available props:

| Name       | Type           | Description  |
| ---------- | ------------- | ----- |
| items | arrayOf(text (string), path (string)) | Required. Items for text label and path |
| onSelect | func | Callback function |
| selectedPage | object | Optional. Selected item on render |
