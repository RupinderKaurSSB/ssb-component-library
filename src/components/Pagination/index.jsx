import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import uuid from 'uuid/v4'
import { ChevronLeft, ChevronRight } from 'react-feather';
import { ssbGreen4, ssbWhite } from '../../style/colors';
import { roboto } from '../../style/mixins';

const Pagination = ({
	items, onSelect, selectedPage,
}) => {
	const [selected, setSelected] = useState(selectedPage || items[0]);
	const [currentButtons, updateCurrentButtons] = useState([{}]);
	const [showLeftDots, updateLeftDots] = useState(false);
	const [showRightDots, updateRightDots] = useState(true);

	useEffect(() => {
		const showItems = items;
		if (items.length < 8) {
			updateLeftDots(false);
			updateRightDots(false);
			updateCurrentButtons(items);
		} else if (items.indexOf(selected) < 7) {
			updateLeftDots(false);
			updateCurrentButtons(showItems.slice(0, 8));
			if (items.length > currentButtons.length) {
				updateRightDots(true);
			}
		} else if (items.indexOf(selected) > (items.length - 7)) {
			updateCurrentButtons(showItems.slice((showItems.length - 8), showItems.length));
			updateRightDots(false);
			updateLeftDots(true);
		} else {
			updateCurrentButtons(showItems.slice((showItems.indexOf(selected) - 3), (showItems.indexOf(selected) + 3)));
			updateLeftDots(true);
			updateRightDots(true);
		}
	}, [selected]);

	const handleSelection = sel => {
		setSelected(sel);
		onSelect(sel);
	};

	return (
		<PaginationWrapper>
			<DirectionButton onClick={() => handleSelection(items[items.indexOf(selected) - 1])}>
				<ChevronLeft className="icon" size={18} />
				<span>Forrige</span>
			</DirectionButton>
			<NavButton
				selected={items[0] === selected}
				onClick={() => handleSelection(items[0])}
			>{items[0].text}
			</NavButton>
			{showLeftDots && <DottedIndicator>...</DottedIndicator>}
			{currentButtons && currentButtons.map(item => (
				item !== items[0] && item !== items[items.length - 1] && (
					<NavButton
						selected={item === selected}
						onClick={() => handleSelection(item)}
						key={uuid}
					>{item.text}
					</NavButton>
				)
			))}
			{showRightDots && <DottedIndicator>...</DottedIndicator>}
			<NavButton
				selected={items[items.length - 1] === selected}
				onClick={() => handleSelection(items[items.length - 1])}
			>{items[items.length - 1].text}
			</NavButton>
			<DirectionButton onClick={() => handleSelection(items[items.indexOf(selected) + 1])}>
				<span>Neste</span>
				<ChevronRight className="icon" size={18} />
			</DirectionButton>
		</PaginationWrapper>
	);
};

const PaginationWrapper = styled.nav`
	align-items: center;
  display: flex;
  ${roboto}
`;

const DirectionButton = styled.div`
	align-items: center;
	border: 1px solid transparent;
	cursor: pointer;
	display: flex;
	height: 30px;
	justify-content: center;
	width: 80px;
	
	.icon {
		color: ${ssbGreen4};
	}
	
	&:hover {
		border: 1px solid ${ssbGreen4};
		border-radius: 2px;
		color: ${ssbGreen4};
		font-weight: bold;
	}
`;

const NavButton = styled.div`
	align-items: center;
	background: ${props => (props.selected ? ssbGreen4 : ssbWhite)};
	border-radius: 2px;
	color: ${props => (props.selected ? ssbWhite : ssbGreen4)};
	cursor: pointer;
	display: flex;
	height: 30px;
	justify-content: center;
	margin: 0 10px;
	width: 30px;
	
	&:hover {
		border: 1px solid ${ssbGreen4};
		border-radius: 2px;
		font-weight: ${props => (props.selected ? 'inherit' : 'bold')};
    }
`;

const DottedIndicator = styled.div`
	align-items: center;
	display: flex;
	justify-content: center;
	margin: 0 10px;
	width: 30px;
`;

Pagination.defaultProps = {
	onSelect: () => {},
};

Pagination.propTypes = {
	items: PropTypes.arrayOf(PropTypes.object).isRequired,
	onSelect: PropTypes.func,
	selectedPage: PropTypes.object,
};

export default Pagination;
