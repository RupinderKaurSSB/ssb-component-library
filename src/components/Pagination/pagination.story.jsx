import React from 'react';
import { storiesOf } from '@storybook/react';
import centered from '@storybook/addon-centered/react';
import { ChevronLeft, ChevronRight } from 'react-feather';
import Pagination from './index';

const items = [
	{ text: '1', path: '/1' },
	{ text: '2', path: '/2' },
	{ text: '3', path: '/3' },
	{ text: '4', path: '/4' },
	{ text: '5', path: '/5' },
	{ text: '6', path: '/6' },
	{ text: '7', path: '/7' },
	{ text: '8', path: '/8' },
	{ text: '9', path: '/9' },
	{ text: '10', path: '/10' },
	{	text: '11', path: '/11'	},
	{	text: '12', path: '/12'	},
	{	text: '13', path: '/13'	},
	{	text: '14', path: '/14'	},
	{	text: '15', path: '/15'	},
	{	text: '16', path: '/16'	},
	{	text: '17', path: '/17'	},
	{	text: '18', path: '/18'	},
	{	text: '19', path: '/19'	},
	{	text: '20', path: '/20'	},
];

storiesOf('Pagination', module).addDecorator(centered)
	.add('Default', () => (
		<div>
			<Pagination items={items} selectedPage={items[4]} />
		</div>
	))
	.add('Html', () => (
		<nav class="ssb-pagination">
			<button class="direction-button previous">
				<ChevronLeft size={18} className="chevron-icon" />Forrige
			</button>
			<button class="nav-button-square">1</button>
			<button class="nav-button-square 2">2</button>
			<button class="nav-button-square 3">3</button>
			<button class="nav-button-square 4">4</button>
			<button class="nav-button-square 5 selected">5</button>
			<button class="nav-button-square 6">6</button>
			<button class="nav-button-square 7">7</button>
			<button class="nav-button-square 8">8</button>
			<div class="dotted-indicator">...</div>
			<button class="nav-button-square">20</button>
			<button class="direction-button next">
				Neste<ChevronRight size={18} className="chevron-icon" />
			</button>
		</nav>
	));
