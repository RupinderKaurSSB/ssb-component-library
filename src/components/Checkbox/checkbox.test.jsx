import React from 'react';
import {shallow} from 'enzyme';
import Checkbox from './index';
import Pagination from '../Pagination'

describe('Checkbox component', () => {
	test('Matches the snapshot', () => {
		const wrapper = shallow(<Checkbox value="item">Item</Checkbox>);
		expect(wrapper).toMatchSnapshot();
	});
	//TODO: onchange test
});
