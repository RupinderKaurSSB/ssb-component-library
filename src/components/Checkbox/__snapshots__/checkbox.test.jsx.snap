// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component Matches the snapshot 1`] = `
<div
  className="ssb-checkbox"
>
  <input
    id="item"
    onChange={[Function]}
    tabIndex={0}
    type="checkbox"
    value="item"
  />
  <label
    className="checkbox-label"
    htmlFor="item"
  >
    Item
  </label>
</div>
`;
